job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
@elimisteve Sorry, I should have been more clear. There is a deadlock if its not unlocked. `DeleteFromParentJobs` uses the cache to perform a `cache.Get()` - https://github.com/ajvb/kala/pull/60/files#diff-bdea023396d54a005d07cc0d83e2c058R257